Development/nmos/authorization_handlers.cpp

```diff
@@ -0,0 +1,378 @@
+                os.close();
+            }
+        }
+
+        // construct callback to load a table of authorization server uri vs authorization client metadata from file based on settings seed_id
+        // it is not required for scopeless OAuth 2.0 client (client not require to access any protected APIs)
+        load_authorization_clients_handler make_load_authorization_clients_handler(const nmos::settings& settings, slog::base_gate& gate)
+        {
+            return [&]()
+            {
+                // obtain client metadata from the safe, permission-restricted, location in the non-volatile memory, e.g. a file
+                // Client metadata SHOULD consist of the client_id, client_secret, client_secret_expires_at, client_uri, grant_types, redirect_uris, response_types, scope, token_endpoint_auth_method
+                auto filename = nmos::experimental::fields::seed_id(settings) + U(".json");
+                slog::log<slog::severities::more_info>(gate, SLOG_FLF) << "Load authorization clients from non-volatile memory: " << filename;
+
+                return load_authorization_clients_file(filename, gate);
+            };
+        }
+
+        // construct callback to save the authorization server uri vs authorization client metadata table to file, using seed_id for the filename
+        // it is not required for scopeless OAuth 2.0 client (client not require to access any protected APIs)
+        save_authorization_client_handler make_save_authorization_client_handler(const nmos::settings& settings, slog::base_gate& gate)
+        {
+            return [&](const web::json::value& authorization_client)
+            {
+                // Client metadata SHOULD be stored in a safe, permission-restricted, location in non-volatile memory in case of a device restart to prevent re-registration.
+                // Client secrets SHOULD be encrypted before being stored to reduce the chance of client secret leaking.
+                // see https://specs.amwa.tv/is-10/releases/v1.0.0/docs/4.2._Behaviour_-_Clients.html#client-credentials
+                const auto filename = nmos::experimental::fields::seed_id(settings) + U(".json");
+                slog::log<slog::severities::more_info>(gate, SLOG_FLF) << "Save authorization clients to non-volatile memory: " << filename;
+
+                update_authorization_clients_file(filename, authorization_client, gate);
+            };
+        }
+
+        // construct callback to start the authorization code flow request on a browser
+        // it is required for those OAuth client which is using the Authorization Code Flow to obtain the access token
+        // note: as it is not easy to specify the 'content-type' used in the browser programmatically, this can be easily
+        // fixed by installing a browser header modifier
+        // such extension e.g. ModHeader can be used to add the missing 'content-type' header accordingly
```

Comment:
```suggestion
        // this is required for OAuth clients which use Authorization Code Flow to obtain the access token
        // note: as it is not easy to specify the 'content-type' used in the browser programmatically, this can be easily
        // fixed by installing a browser header modifier
        // extensions such as ModHeader can be used to add the missing 'content-type' header:
```